src/OrchardCore.Modules/OrchardCore.Lucene/LuceneQueryDTO.cs

```diff
@@ -0,0 +1,13 @@
+using Newtonsoft.Json;
+
+namespace OrchardCore.Lucene
+{
+
+   [JsonConverter(typeof(LuceneQueryDTOConverter))]
+   public class LuceneQueryDTO
+   {
+      public string IndexName { set; get; }
+      public string Query { set; get; }
+      public string Parameters { set; get; }
```

Comment:
why does this need a custom json convertor?

Can the model not just be `Dictionary<string,object>` here instead?

Reply:
Yes , it will work this will get rid of the json converter

> :    public async Task<IActionResult> ContentPost(Dictionary<string,object> query)
        {

            LuceneQueryDTO queryDto=new LuceneQueryDTO(){
                IndexName = query["indexName"].ToString(),
                Query = query["query"].ToString(),
                Parameters = query["parameters"].ToString()
            };

Reply:
Even with a DTO, I don't understand the need of a custom converter.

Reply:
ok, The dto is used by the get and the post. For the get it is simply mapping from the query as strings
for the post the json load is three parts the indexName is string the query and the parameters are dynamic json that's is why we can not have the default binding to dto object works out of the box. we needed to load the json as whole objects then parse different properties like what @deanmarcussen suggested . or using custom converter
@jtkech pointed me to the contentItem api and i saw how it worked with half static half dynamic content of content items
if I did the same this would look like :
 public class LuceneQueryDTO
   {
      public string IndexName { set; get; }
      public dynamic Query { set; get; }
      public dynamic Parameters { set; get; }
   }
for the sake of get endpoint i avoided doing this because we would need different inputs
I am not sure which way to go . I prefer what @jtkech pointed because it keeps the code clean and explain the intent

Reply:
> for the sake of get endpoint i avoided doing this because we would need different inputs

I would suggest making the endpoints different (that is for me the point of what you are doing here, is seperating out the differences between a `Get` and a `Post` )

```
public class LuceneQueryModel
{
  public string IndexName { set; get; }
  public string Query { set; get; }
  public Dictionary<string, object> Parameters { set; get; }
}

Other option you could also do is leave the `Get` and `Post` exactly as they are, and make a new endpoint `Put` with the new binding. Non breaking that way. and Elastic Search uses `Put` rather than `Post`, so is in keeping to have a `Put` endpoint for search (`Post` suggests that it's creating something here, when really it's just a query, that is idempotent)

Reply:
POST means "create new" as in "Here is the input for creating a user, create it for me". PUT means "insert, replace if already exists" as in "Here is the data for user 5".

So neither are good for retrieving the data. This is why I'm not sure why we want it to work with a POST when this should work only with a GET if it retrieves only data.

Reply:
Okay, I need to see the usage of POST and PUT on ElasticSearch. Because, here, this ApiController is meant to retrieve the data from a Lucene Query alias ElasticSearch query. Which means we should be able to pass a JSON Query in either a form or a querystring input/param. Though, there is no difference here in this PR about doing a GET or a POST.

Here is the difference clearly defined on this page for ElasticSearch : 

https://www.elastic.co/guide/en/elasticsearch/guide/current/index-doc.html#_using_our_own_id
https://www.elastic.co/guide/en/elasticsearch/guide/current/index-doc.html#_autogenerating_ids

And of course a GET is only a GET...